docs/cicd-with-bicep.md

```diff
@@ -73,5 +68,88 @@ jobs:
             az account show
             az deployment group create -f ./main.json -g ${{ env.AZURE_RESOURCE_GROUP }}
 ```
+## Azure DevOps Pipeline
+
+```yaml
+trigger:
+- main
+name: 'bicep build and deploy'
+
+variables:
+  vmImageName: 'ubuntu-latest'
+  workingDirectory: '$(System.DefaultWorkingDirectory)/'
+  geoLocation: 'West Europe'
 
-Instead of installing the Bicep CLI manually, you may instead want to use the [community-maintained github action](https://github.com/marketplace/actions/bicep-build) from [@justinyoo](https://github.com/justinyoo) that can run `bicep build` on your behalf.
+  azureServiceConnection: 'My-Azure-DevOps-ServicePrincipalName'
+  subscriptionId: 'My-Subscription-Id'
+  AZURE_RESOURCE_GROUP: 'myResourceGroupName'
+  
+stages:
+- stage: Build
+  displayName: Build
+      
+  jobs:
+  - job: Build
+    displayName: Validate and Publish
+    pool:
+     vmImage: $(vmImageName)
+      
+    steps:
+      - task: AzureCLI@2
+        displayName: Build ARM Template from bicep file
+        inputs:
+          azureSubscription: '$(azureServiceConnection)'
+          scriptType: bash
+          scriptLocation: inlineScript
+          inlineScript: |
+            az --version
+            az bicep build --files ./main.bicep
```

Comment:
I think you need to first call ‘az bicep install’ first.

Reply:
you actually don't have to (though you can). If you attempt to create a deployment with a `.bicep` file and bicep is not installed, we will install it automatically

Reply:
Ok, right. I just tested and it does not work with `az bicep version` but it does with `build`. Apologies for mixup.

Reply:
Correct:
using `az bicep build` will download the CLI if needed, and will then transpile it if needed. (See screenshot) below. So no need for the curl and extension hassle.
Although you still need to transpile to JSON to actually deploy this.

![image](https://user-images.githubusercontent.com/57876192/111531978-7f2c7a00-8765-11eb-9abd-4ea6b79746ef.png)